### PR TITLE
7 implimented POST/api/articles/article_id/comments endpoint

### PR DESCRIPTION
**Pull request by DeanTS96**
Unsure if I need to test for empty message and user not existing in database

Also is there a better way I could impliment the functionality in the model instead of having to wait for a query to then go to the next with a then chain. using promise.all wouldn't allow me to get the specific error and send a specific response.

### CODE REVIEW

File: models/comments.models.js [hunk 1]
```diff
@@ -10,4 +9,22 @@ function fetchCommentsById(params) {
...
+    if(!comment.body.length) return Promise.reject({status: 400, msg: 'empty comment'});
+    return db.query(`SELECT * FROM users WHERE username = $1`, [comment.username])
+    .then(({rows}) => {
+        if(!rows.length) return Promise.reject({status: 404, msg: 'user doesn\'t exist'})
+        return db.query(`SELECT * FROM articles WHERE article_id = $1`, [params.article_id])
+    })
+    .then(({rows}) => {
+        if(!rows.length) return Promise.reject({status: 404, msg: 'article doesn\'t exist'});
+        return db.query(`
+        INSERT INTO comments (body, author, created_at, article_id, votes)
```
Comment (two-triangles) [inline models/comments.models.js]: you should not need to post into the created_at and votes rows as these have default values(or at least should do!) in your database schema - these can be removed.

File: models/comments.models.js [hunk 2]
```diff
@@ -10,4 +9,22 @@ function fetchCommentsById(params) {
     });
 }
 
-module.exports = {fetchCommentsById};
\ No newline at end of file
+function addCommentByArticleId(comment, params) {
+    if(!comment.body.length) return Promise.reject({status: 400, msg: 'empty comment'});
+    return db.query(`SELECT * FROM users WHERE username = $1`, [comment.username])
```
Comment (two-triangles) [inline models/comments.models.js]: So if you try and insert a comment with a user who does not exist in the database, PSQL will give you an error back, with a specific code that you can utilise in your error handling :)
